### PR TITLE
KEYCLOAK-10189 Possibility to use Gitlab for build keycloak sources

### PR DESCRIPTION
**Pull request by Kilian22**
I add the possibility to use for exemple Gitlab repo when you want to build a new keycloak image from a non official repository.

### CODE REVIEW

**Comment by slaskawi** [comment]
Even though I like this change, but we keep Keycloak sources in Github in general. So I'm not sure if this is something that should belong to a custom user repo rather than this one. 

But let's see what @stianst thinks about it.

File: server/Dockerfile [hunk 1]
```diff
@@ -12,6 +12,7 @@ ENV LANG en_US.UTF-8
 
 ARG GIT_REPO
 ARG GIT_BRANCH
+ARG GIT_HOST
```
Comment (stianst) [inline server/Dockerfile]: Set default value to https://github.com here rather than in build-keycloak.sh.

File: server/README.md [hunk 2]
```diff
@@ -335,6 +335,12 @@ It is possible to build Keycloak from a GitHub repository instead of downloading
 
 This will clone the repository then build Keycloak from source. If you don't include GIT_BRANCH it will use the `master` branch.
 
+You can also use another solution of repository like Gitlab using GIT_HOST option.
```
Comment (stianst) [inline server/README.md]: No need to document this. I really don't think anyone else will use it.

File: server/tools/build-keycloak.sh [hunk 3]
```diff
@@ -9,20 +9,25 @@ if [ "$GIT_REPO" != "" ]; then
         GIT_BRANCH="master"
     fi
 
+if [ "$GIT_HOST" == "" ]; then
```
Comment (stianst) [inline server/tools/build-keycloak.sh]: Remove, use default value for build arg as mentioned above

**Comment by stianst** [comment]
Closing due to lack of feedback

**Pull request closed without merging**